### PR TITLE
Additional engine ES6 classes

### PR DESCRIPTION
**Pull request by mvaligursky**
layer
forward-renderer
particle-system

### CODE REVIEW

File: src/math/mat4.js [hunk 1]
```diff
@@ -1270,6 +1270,24 @@ class Mat4 {
...
+        dst[2] = src[2];
+
+        dst[3] = src[4];
+        dst[4] = src[5];
+        dst[5] = src[6];
+
+        dst[6] = src[8];
+        dst[7] = src[9];
+        dst[8] = src[10];
+    }
```
Comment (willeastcott) [inline src/math/mat4.js:1289]: Not sure this quite matches the naming convention of:

```
pc.Thing#setFromOtherThing
```

e.g. `pc.Quat#setFromMat4`

I mean, we also have `pc.Quat#toString` say. Ugh, I hate these sorts of consistency problems....

Comment (mvaligursky) [reply src/math/mat4.js:1289]: ok, I'll change it to Mat3.setFrom(mat4) .. that seems more consistent?
probably easier to read as well, as class that its called on is the target.

Comment (willeastcott) [reply src/math/mat4.js:1289]: Yeah, I think I prefer that.

Comment (mvaligursky) [reply src/math/mat4.js]: and done. looks better.

**Review by willeastcott — APPROVED** [review]

**Pull request merged**